src/controllers/router-store/index.tsx

```diff
@@ -200,6 +212,38 @@ const actions: AllRouterActions = {
 
     return { query, route, match };
   },
+
+  pushQueryParam: params => ({ getState }) => {
+    const { query: existingQueryParams, history, location } = getState();
+    const updatedQueryParams = { ...existingQueryParams, ...params };
+    // remove null/undefined keys
```

Comment:
I left a comment on the test, but we should discuss if this behavior is desired. Was there a request to remove params that have a `null` or `undefined` value?

Reply:
Yes, we should allow consumers to remove query parameters. However we should pick one: I'm worried that by allowing both `null` or `undefined` we complicate our lives.

Like what is the outcome of `useQueryParam('foo')` or `usePathParam('foo')` when `foo` is not defined? Is that `null` or `undefined`? We should be clear about that, consistent and we should add a couple of tests around it ensuring that the values returned are correct.

Reply:
I'm happy with that. Supporting `null` and `undefined` does seem to complicate it for no good reason.